tests/cases/fourslash/autoImportPackageJsonImportsLength1.ts

```diff
@@ -0,0 +1,18 @@
+/// <reference path="fourslash.ts" />
+
+// @module: nodenext
+
+// @Filename: /package.json
+//// {
+////   "imports": {
+////     "#*": "./src/*.ts"
```

Comment:
I'm confused; isn't this invalid as it's definitely going to crash at runtime? Or is that what is being tested?

Reply:
From talking with @andrewbranch: compiler already has some special re-mapping logic to map `imports` that describe runtime to their respective source files written in TS. This particular example isn't about that though but it's still valid if you don't use TS for emit. For example, we are just bundling whatever is in `imports` so they are not observable by the runtime - the checker has to resolve them within source files though and this works.

Reply:
An example of the latter in the wild: https://github.com/Andarist/react-textarea-autosize/blob/eb4e4f1b397e55b436dfef349d8b3c8c9733a07e/package.json#L59-L69

Reply:
I would say this is bad practice without `allowImportingTsExtensions`, but `allowImportingTsExtensions` doesn’t prevent it because the file extension is specified in the package.json, not in the source code. This is consistent with `"main": "./index.ts"`—obviously sketchy, but not something we go out of our way to fail on.